server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -24,72 +25,128 @@
-        builder.endObject();
-
-        builder.startObject(Fields.REMOTE_REFRESH_SIZE_IN_BYTES);
-        builder.field(SubFields.LAST_SUCCESSFUL, remoteSegmentUploadShardStats.lastSuccessfulRemoteRefreshBytes);
-        builder.field(SubFields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesMovingAverage);
-        builder.endObject();
-
-        builder.startObject(Fields.UPLOAD_LATENCY_IN_BYTES_PER_SEC);
-        builder.field(SubFields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesPerSecMovingAverage);
-        builder.endObject();
-        builder.startObject(Fields.REMOTE_REFRESH_LATENCY_IN_MILLIS);
-        builder.field(SubFields.MOVING_AVG, remoteSegmentUploadShardStats.uploadTimeMovingAverage);
+        builder.startObject();
+        builder.startObject(Fields.ROUTING);
+        builder.field(RoutingFields.STATE, shardRouting.state());
+        builder.field(RoutingFields.PRIMARY, shardRouting.primary());
+        builder.field(RoutingFields.NODE_ID, shardRouting.currentNodeId());
         builder.endObject();
+        builder.startObject(Fields.SEGMENT);
+        if (!shardRouting.primary()) {
+            builder.startObject(SubFields.DOWNLOAD);
+            builder.field(DownloadStatsFields.LAST_DOWNLOAD_TIMESTAMP, remoteSegmentShardStats.lastDownloadTimestampMs);
+            builder.startObject(DownloadStatsFields.TOTAL_SYNCS_FROM_REMOTE)
+                .field(SubFields.STARTED, remoteSegmentShardStats.totalDownloadsStarted)
+                .field(SubFields.SUCCEEDED, remoteSegmentShardStats.totalDownloadsSucceeded)
+                .field(SubFields.FAILED, remoteSegmentShardStats.totalDownloadsFailed);
+            builder.endObject();
+            builder.startObject(DownloadStatsFields.TOTAL_FILE_DOWNLOADS_IN_BYTES)
+                .field(SubFields.STARTED, remoteSegmentShardStats.downloadBytesStarted)
+                .field(SubFields.SUCCEEDED, remoteSegmentShardStats.downloadBytesSucceeded)
+                .field(SubFields.FAILED, remoteSegmentShardStats.downloadBytesFailed);
+            builder.endObject();
+            builder.startObject(DownloadStatsFields.DOWNLOAD_SIZE_IN_BYTES)
+                .field(SubFields.LAST_SUCCESSFUL, remoteSegmentShardStats.lastSuccessfulSegmentDownloadBytes)
+                .field(SubFields.MOVING_AVG, remoteSegmentShardStats.downloadBytesMovingAverage);
+            builder.endObject();
+            builder.startObject(DownloadStatsFields.DOWNLOAD_SPEED_IN_BYTES_PER_SEC)
+                .field(SubFields.MOVING_AVG, remoteSegmentShardStats.downloadBytesPerSecMovingAverage);
+            builder.endObject();
+            builder.startObject(DownloadStatsFields.DOWNLOAD_LATENCY_IN_MILLIS)
+                .field(SubFields.MOVING_AVG, remoteSegmentShardStats.downloadTimeMovingAverage);
+            builder.endObject();
+            builder.endObject();
+        } else {
+            builder.startObject(SubFields.DOWNLOAD);
+            builder.endObject();
```

Comment:
the start can be moved to before the If start and the end can be moved after the If. We would not need the else.

Reply:
same can be used for Upload as well.